platform/core/src/classes/MetadataProvider.js

```diff
@@ -441,6 +441,7 @@ class MetadataProvider {
   }
 
   getUIDsFromImageID(imageId) {
+    if (!imageId) throw new Error('MetadataProvider::Empty imageId');
```

Comment:
Now that we are throwing this `Error` do we need to handle it somewhere?

Reply:
It was previously a NPE, so this just gets an error message when it happens.  Yes, it needs to be handled, but I haven't gotten a hold of a good stack trace that shows what is happening when it is null.  I think it is a race condition.